src/web/assets/cp/src/js/ElementEditor.js

```diff
@@ -1464,16 +1469,35 @@ Craft.ElementEditor = Garnish.Base.extend(
 
             // Make the first tab visible if no others are
             if (!$allTabContainers.filter(':not(.hidden)').length) {
-              $allTabContainers.first().removeClass('hidden');
+              if (this.isFullPage) {
+                Craft.cp.selectTab($allTabContainers.first());
+              }
```

Comment:
This change is preventing any content from showing when the originally-selected tab isn’t present after the autosave (e.g. when switching entry types, or when a tab condition comes into effect.) `$allTabContainers` is for the content  container elements, not the tabs themselves, so not something you would pass into `selectTab()`.